### DISCUSSION TITLE
wireguard inbound reserved bytes

### DISCUSSION THREAD

**Discussion by siq0o** [XTLS/Xray-core#2954] [Ideas]
as far as I'm concerned, it is possible to have a successful handshake using the reserved bytes, in places where the wg protocol is blocked by dpi (similar to warp). supporting the same method for inbound connections could be very useful.

**Comment by us254** [comment]
https://xtls.github.io/en/config/outbounds/wireguard.html#outboundconfigurationobject

```
{
  "outbounds": [
    {
      "protocol": "wireguard",
      "settings": {
        "secretKey": "PRIVATE_KEY",
        "address": [
          "172.16.0.2/32",
          "2606:4700::/128"
        ],
        "peers": [
          {
            "endpoint": "162.159.192.1:2408",
            "publicKey": "bmXOC+F1FxEMF9dyiK2H5/1SUtzH0JuVo51h2wPfgyo=",
            "preSharedKey": "PRE_SHARED_KEY_PLACEHOLDER",
            "keepAlive": 0,
            "allowedIPs": [
              "0.0.0.0/0",
              "::/0"
            ]
          }
        ],
        "mtu": 1420,
        "reserved": [
          1,
          2,
          3
        ],
        "workers": 2,
        "domainStrategy": "ForceIPv6v4"
      },
      "tag": "warp"
    }
  ],
  "DNS": {
    "servers": [
      {
        "address": "https://1.1.1.1/dns-query",
        "domains": [
          "geosite:openai"
        ],
        "skipFallback": true,
        "queryStrategy": "UseIPv6"
      }
    ],
    "queryStrategy": "UseIP"
  }
}
```

**reserved [ number ]**

**Wireguard Reserved Bytes.**

**Xray-core v1.8.0 New parameter.**
When connecting to warp via wireguard, due to cloudflare limitations, **_some IPs in Hong Kong and Los Angeles need to have a reserved value in order to connect successfully._**
The value of reserved can be obtained using third-party tools such as `warp-reg, warp-reg.sh.。`

https://github.com/badafans/warp-reg

https://github.com/chise0713/warp-reg.sh

**Reply by siq0o** [reply]
I'm aware of the outbound wireguard settings.
> supporting the same method for inbound connections could be very useful.

**Comment by APT-ZERO** [comment]
Really? Will that bypass firewall detecting Wireguard protocol? I don't think so
Wireguard protocol is not made to bypass any firewall, It's getting easily detected by all firewalls because there is no obfuscation

**Reply by siq0o** [reply]
True, it won't obfuscate it.
The firewall only checks the first packet (at least in Iran). if it's a handshake initiation, drops it. check out [this article](https://www.ntkernel.com/how-to-bypass-egypts-wireguard-ban/)
